spinnaker-monitoring-daemon/spinnaker-monitoring/spectator_metric_transformer.py

```diff
@@ -449,6 +457,11 @@ def composite_func(value):
       else:
         matched_values = matched.groups()
 
+      if not allof:
+        # Reduce oneof_regex to a single matched value
+        matched_values = [value for value in matched_values
```

Comment:
Is this actually reducing to the first matched value?  It looks to just be filtering out `None`

Reply:
I think it is removing the None and leaving all non-None.
The assumption is that the regex contains exactly one non-None value, so this would be it. If not then this will result in an error (in the #465 general check) because the spec is wrong.